File: uva/Leading_and_Trailing/Leading_and_Trailing.py
def pegar3ultimos(n):
    temp = str(n)
    result = ""
    
    cont = 0
    for i in range(len(temp)-1,-1,-1):
        if cont >= 3:
            break
        result += temp[i]
        cont+=1
    
    result2 = ""    
    for i in range(len(result) - 1, -1, -1):
        result2 += result[i]
        
    return int(result2)


t = int(input())

while(t > 0):
    t-=1
    n, k = map(int, input().split())

    ans1 = 1
    ans2 = 1
    tresultimos = pegar3ultimos(n)
    for i in range(k):    
        ans1 = (ans1 * n) % 10000000000000000000000000000000000000000000000000000000
        ans2 = (ans2 * tresultimos)

    ans2 = pegar3ultimos(ans2)
    
    ans1 = str(ans1)
    ans2 = str(ans2)
    for i in range(3):
        print(ans1[i], end="")

    print("...", end="")
    
    for i in range(3):
        print(ans2[i], end="")

    print()

File: uva/Leading_and_Trailing/Leading_and_Trailing.java
import java.util.Scanner;
import java.math.BigDecimal;
public class Leading_and_Trailing{

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t;
        t = sc.nextInt();

        while(t > 0){
            t--;

            BigDecimal n;
            int k;
            n = sc.nextBigDecimal();
            k = sc.nextInt();

            BigDecimal ans1 = BigDecimal.ONE;
            BigDecimal ans2 = BigDecimal.ONE;

            BigDecimal tresultimos = new BigDecimal(pegar3ultimos(n));

            for(int i = 0; i < k ; i++){    
                ans1 = ans1.multiply(n).remainder(BigDecimal.valueOf(1000000000));
                ans2 = ans2.multiply(tresultimos);
            }

            String ans1String = ans1.toString();
            String ans2String = ans2.toString();

            for(int i = 0; i < 3; i++){
                System.out.printf("%c", ans1String.charAt(i));
            }            

            System.out.printf("...");

            
            for(int i = 0; i < 3; i++){
                System.out.printf("%c", ans2String.charAt(i));
            }            

            System.out.printf("\n");

        }
        
    }
    static int pegar3ultimos(BigDecimal n){
        String temp = n.toString();
        String result = "";

        int cont = 0;
        for(int i = temp.length()-1; i >= 0 && cont < 3; i--){
            result += temp.charAt(i);
            cont++;
        }

        String result2 = "";
        for(int i = result.length() -1; i >= 0; i--){
            result2 += result.charAt(i);
        }
        return Integer.valueOf(result2);

    }
}